xwiki-platform-tools/xwiki-platform-tool-configuration-resources/src/main/resources/xwiki.properties.vm

```diff
@@ -878,6 +878,26 @@ distribution.automaticStartOnWiki=$xwikiPropertiesAutomaticStartOnWiki
 #-# issues
 # security.requiredRights.protection=warning
 
+#-------------------------------------------------------------------------------------
+# Query
+#-------------------------------------------------------------------------------------
+
+#-# [Since 17.0.0RC1]
+#-# [Since 16.10.2]
+#-# [Since 15.10.16]
+#-# [Since 16.4.6]
+#-# While HQL queries executed by users without programming right are validated. It's sometimes necessary to
+#-# customize the behavior of the standard validator because it might be too strict (or not strict enough).
+#-# The following properties allows to give a list of (Java) regular expression.
+#-# In case both safe and unsafe patterns match the query, the priority goes to considering it unsafe.
+#-#
+# query.hql.unsafe=.*some native syntax.*
+#-# 
+#-# Note: be very careful to not use too large regular expression when adding a safe pattern as it could introduce a vulnerability.
+#-# 
+# query.hql.safe=select pro1, prop2 from CustomTable
+# query.hql.safe=select\\s+((prop1|prop2|prop3)\\s*,?\\s*)+\\s+from MyCustomTable
```

Comment:
Some small typos and ideas of improvements:

1) Grammar error. Proposal:

   > While HQL queries executed by users without programming right are already validated, it's sometimes necessary to [...]

2) It would be even nicer to give a real use case example.

3) Small grammar error. s/allows/allow (since there's more than 1 property)

4)  Small grammar error. s/expression/expressions (it's plural since there's a list).

5) s/to considering it unsafe/to the unsafe pattern  or s/to applying the unsafe pattern

6) s/select pro1/select prop1

Thx

Reply:
> It would be even nicer to give a real use case example.

To me, the example which indicate a way to allow selecting from a custom table you introduce and consider safe to read is a real use case. Any example that would be based on a standard table would probably mean that there is a bug in the default validator.